pkg/types/compactor.go

```diff
@@ -0,0 +1,92 @@
+package types
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+	"time"
+
+	flag "github.com/spf13/pflag"
+	"go.etcd.io/etcd/pkg/types"
+)
+
+// CompactionResult holds the compaction details after a compaction is done
+type CompactionResult struct {
+	// Snapshot               *snapstore.Snapshot
+	Path                   string
+	LastCompactionDuration time.Duration
+}
+
+// CompactionConfig holds the configuration data for compaction
+type CompactionConfig struct {
+	rc *RestorationConfig
```

Comment:
Why not embed `*RestorationConfig` and _inherit_ `Validate` and `AddFlags` as well? Additional behaviour can be done by _overriding_.

Reply:
Done